src/backend/distributed/commands/multi_copy.c

```diff
@@ -2291,13 +2324,53 @@ CitusCopyDestReceiverStartup(DestReceiver *dest, int operation,
 	RecordRelationAccessIfNonDistTable(tableId, PLACEMENT_ACCESS_DML);
 
 	/*
-	 * For all the primary (e.g., writable) nodes, reserve a shared connection.
-	 * We do this upfront because we cannot know which nodes are going to be
-	 * accessed. Since the order of the reservation is important, we need to
-	 * do it right here. For the details on why the order important, see
-	 * the function.
+	 * For all the primary (e.g., writable) remote nodes, reserve a shared
+	 * connection. We do this upfront because we cannot know which nodes
+	 * are going to be accessed. Since the order of the reservation is
+	 * important, we need to do it right here. For the details on why the
+	 * order important, see the function.
```

Comment:
I did not find the comment on EnsureConnectionPossibilityForRemotePrimaryNodes very enlightening on why this is important.

Reply:
Oh, the details of the requirements for the order is on `EnsureConnectionPossibilityForNodeList`, so updating it